bundle/Command/LegacySrcSymlinkCommand.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * File containing the LegacySrcSymlinkCommand class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace eZ\Bundle\EzPublishLegacyBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+class LegacySrcSymlinkCommand extends ContainerAwareCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setName('ezpublish:legacy:symlink_src')
+            ->setDefinition(
+                array(
+                    new InputArgument('src', InputArgument::OPTIONAL, 'The src directory for legacy files', 'src/legacy_files'),
+                )
+            )
+            ->addOption('create', 'c', InputOption::VALUE_NONE, 'Create src folder structure if it does not exists')
+            ->addOption('force', 'f', InputOption::VALUE_NONE, 'Symlink folders even if target already exists')
+            ->setDescription('Installs legacy project settings and design files from "src" to corresponding folders in ezpublish_legacy/')
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> setups and symlinks <info>src/legacy_files</info> stored in your root project for
+any design/settings project files, and symlinks these into <info>ezpublish_legacy/</info> which is installed by composer.
+
+<comment>NOTE: Look towards 'ezpublish:legacybundles:install_extensions' command for how you handle extensions.</comment>
+EOT
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $srcArg = rtrim($input->getArgument('src'), '/');
+        $create = (bool)$input->getOption('create');
+        $force = (bool)$input->getOption('force');
+
+        /**
+         * @var \Symfony\Component\Filesystem\Filesystem
+         */
+        $filesystem = $this->getContainer()->get('filesystem');
+
+        if (!$filesystem->exists($srcArg)) {
```

Comment:
What's the use creating the source folder structure?

Reply:
Migration/Upgrade use case:
- install legacy bridge (will execute this step)
- manually move files over from old to new structure


So no manual steps needed for creating the folders, and it can be explicit referred to in doc where you move them as a default convention

Reply:
How hard is it to type `mkdir -p src/legacy_files/settings` ? :) Doesn't seem that much useful to me if it's documented right, but okay, if you think it can help, sure, but I still don't like that script is running with `-c` automatically *every time*.

Reply:
I'm with @emodric here on having `-c` as default.  Leaving off `-c` will throw the exception and help the users who have not read the docs be aware they can and should migrate their existing config to the new location.

Reply:
PR updated, also added much more output to try to explain the concepts better for user.

better? _(changes done in new commit)_